resources/design-document.md

```diff
@@ -0,0 +1,206 @@
+ 
+email // string 
+gameId // string
+date // string (converted date)
+outcome // enum
+aspect // enum
+gameCharacters // List<PlayerCharacter>
+
+```
+
+```
+
+// PlayerCharacterModel
+
+name // string
+roles // List<Enum>   
+
+```
+
+
+## 6.2 Endpoints
+
+### 6.2.0 _Create GameLog Endpoint_
+- Accepts `POST` requests to `/gamelogs`
+- Accepts data to create a new gameLog with a provided gameId, date, w/l outcome, aspect, hero,
+  and villain. Returns the new GameLog, including a game ID assigned by the service.
+
+### 6.2.1 _Get single GameLog Endpoint_
+- Accepts `GET` requests to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts an email and game ID and returns the corresponding GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.1.1 _Get all GameLogs by player Endpoint_
+- Accepts `GET` requests to `/gamelogs/:`
+- Takes the email from cognito
+- Accepts an email and returns the corresponding list of GameLogModels.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.2 _Delete GameLog Endpoint_
+- Accepts a `DELETE` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts a game ID and returns the corresponding Deleted GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.3 _Update GameLog Endpoint_
+- Accepts a `PUT` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts data to update a gameLog and returns the updated gameLog.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+    - throws `UnauthorizedOwnerException` if attempted to be updated by an unauthorized user.
+
+### 6.2.4 _Get Stats Endpoint_
+- Accepts `GET` requests to `/gamelogs/:`
+- Takes the email from cognito
+- Accepts an email and returns the overall w/l stats for that person's profile.
+
+### 6.2.5 _Get Hero Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:hero`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain hero for that person's profile.
+
+### 6.2.6 _Get Villain Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:villain`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain villain for that person's profile.
+
+### 6.2.7 _Get Aspect Stats Endpoint_
+- Accepts `GET` request to `/gamelogs/:aspect`
+- Takes the email from cognito
+- Accepts an email and returns the w/l stats for a certain aspect for that person's profile.
+
+
+
+# 7. Tables
+
+_Define the DynamoDB tables you will need for the data your service will use. It may be helpful to first think of what
+objects your service will need, then translate that to a table structure, like with the *`Playlist` POJO* versus the
+`playlists` table in the Unit 3 project._
+
+## 7.1 `gamelogs`
+```
+email // partition key, string
+gameId // sort key, String
+date // string (converted date)
+outcome(w/l) // enum
```

Comment:
I think that Dynamo will let you name a field like this but I'd suggest doing something like `outcome_wl` - avoid the parentheses and slashes in field names.

Also while this is an enum in your Java model what will it be in Dynamo?